include/sys/zfs_debug.h

```diff
@@ -61,6 +61,8 @@ extern void __zfs_dbgmsg(char *buf);
 extern void __dprintf(boolean_t dprint, const char *file, const char *func,
     int line, const char *fmt, ...);
 
+extern void __set_error(const char *file, const char *func, int line, int err);
```

Comment:
nit: Can be place it before `__zfs_dbgmsg` which matches the ordering in the `zfs_debug.c`.  No need for an empty line afterwards either.